lib/THC/THCTensorMath2.cu

```diff
@@ -660,3 +660,55 @@ void THCudaTensor_randn(THCState *state, THCudaTensor *r_, THLongStorage *size)
   THCudaTensor_resize(state, r_, size, NULL);
   THCudaTensor_normal(state, r_, 0, 1);
 }
+
+__global__ void THCudaTensor_kernel_cross(float *src_, float *x, float *y, long nelem) {
+  long blockid = blockIdx.x + blockIdx.y * gridDim.x + gridDim.x*gridDim.y*blockIdx.z;
+  long start = 3*(blockid*blockDim.x + threadIdx.x);
+
+  long p1 = start + 0;
+  long p2 = start + 1;
+  long p3 = start + 2;
+  if (p3 > nelem) return;
+  src_[p1] = x[p2]*y[p3] - x[p3]*y[p2];
+  src_[p2] = x[p3]*y[p1] - x[p1]*y[p3];
+  src_[p3] = x[p1]*y[p2] - x[p2]*y[p1];
+}
+
+THC_API void THCudaTensor_cross(THCState *state, THCudaTensor *self, THCudaTensor *x, THCudaTensor *y, int dimension)
+{
+  THAssert(THCudaTensor_checkGPU(state, 3, self, x, y));
```

Comment:
Please use `THCUNN_assertSameGPU` here. It does the same thing, but prints a nice error message if the check fails.